machine_learning/loss_functions.py

```diff
@@ -297,6 +297,52 @@ def mean_squared_logarithmic_error(y_true: np.ndarray, y_pred: np.ndarray) -> fl
     return np.mean(squared_logarithmic_errors)
 
 
+def mean_absolute_percentage_error(y_true: np.ndarray, y_pred: np.ndarray) -> float:
+    """
+    Calculate the Mean Absolute Percentage Error between y_true and y_pred.
+
+    Mean Absolute Percentage Error calculates the average of the absolute
+    percentage differences between the predicted and true values.
+
+    Formula = (Σ|y_true[i]-Y_pred[i]/y_true[i]|)/n
+
+    Source: https://stephenallwright.com/good-mape-score/
+
+    Parameters:
+    y_true (np.ndarray): Numpy array containing true/target values.
+    y_pred (np.ndarray): Numpy array containing predicted values.
+
+    Returns:
+    float: The Mean Absolute Percentage error between y_true and y_pred.
+
+    Examples:
+    >>> y_true = np.array([10, 20, 30, 40])
+    >>> y_pred = np.array([12, 18, 33, 45])
+    >>> mean_absolute_percentage_error(y_true, y_pred)
+    13.124999992135416
+
+    >>> y_true = np.array([1, 2, 3, 4])
+    >>> y_pred = np.array([2, 3, 4, 5])
+    >>> mean_absolute_percentage_error(y_true, y_pred)
+    52.083332977430565
+
+    >>> y_true = np.array([5, 0, 10, 20])
+    >>> y_pred = np.array([5, 0, 9, 15])
+    >>> mean_absolute_percentage_error(y_true, y_pred)
+    8.749999994374999
+    """
+    if len(y_true) != len(y_pred):
+        raise ValueError("The length of the two arrays should be the same.")
+
+    # Calculate the absolute percentage difference between y_true and y_pred
+    # added 1e-9 to avoid division by 0 (smoothing).
+    absolute_percentage_diff = np.abs((y_true - y_pred) / (y_true + 0.00000001))
```

Comment:
```suggestion
    y_true = np.where(y_true == 0, epsilon, y_true)
    absolute_percentage_diff = np.abs((y_true - y_pred) / y_true)
```
1. Please add `epsilon` as an optional parameter to the function with the default value `1e-9`. This way, the user can customize the buffer value.
2. Let's use `np.where` to replace all 0 values with `epsilon` instead of simply adding `epsilon` to every entry. This way, unproblematic values can remain unchanged.